### PR TITLE
JAMES-1822 Introduce docker-compose

### PR DESCRIPTION
**Pull request by aduprat**

### CODE REVIEW

File: README.txt [hunk 1]
```diff
@@ -7,6 +7,33 @@ This is the parent module for all Apache James artifacts. It contains useful val
...
 
+How to try James
+================
+Requirements: docker & docker-compose installed.
+
+When try James by this way, you will use the last state of James.
+It will be configured to run with Cassandra & ElasticSearch.
+All those three components will be started with a single command.
+
+Firt, you need to build the docker-compose image:
```
Comment (rouazana) [inline README.txt]: First

Comment (aduprat) [inline README.txt]: done

File: README.txt [hunk 2]
```diff
@@ -7,6 +7,33 @@ This is the parent module for all Apache James artifacts. It contains useful val
...
 
+How to try James
+================
+Requirements: docker & docker-compose installed.
+
+When try James by this way, you will use the last state of James.
+It will be configured to run with Cassandra & ElasticSearch.
+All those three components will be started with a single command.
+
+First, you need to build the docker-compose image:
```
Comment (mbaechler) [inline README.txt]: I think it builds docker images, nothing special about docker-compose

File: README.txt [hunk 3]
```diff
@@ -7,6 +7,33 @@ This is the parent module for all Apache James artifacts. It contains useful val
...
+$ docker-compose -f dockerfiles/run/docker-compose.yml build
+
+Then, you just have to start the services:
+$ docker-compose -f dockerfiles/run/docker-compose.yml up
+
+Wait a few seconds in order to have all those services up, you will see a such log when James is available:
+james           | Started : true
+
+Then, a default domain has been created: james.org
```
Comment (mbaechler) [inline README.txt]: could we change is to something like james.local ?

File: README.txt [hunk 4]
```diff
@@ -7,6 +7,33 @@ This is the parent module for all Apache James artifacts. It contains useful val
...
+$ docker exec james java -jar /root/james-cli.jar -h 127.0.0.1 -p 9999 listdomains
+james.org
+
+James will respond to IMAP port 1143 and SMTP port 1025.
```
Comment (mbaechler) [inline README.txt]: why not 143 and 25 ?

File: dockerfiles/run/docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,29 @@
+version: '2'
+
+services:
+
+  james:
+    depends_on:
+      - elasticsearch
+      - cassandra
+    entrypoint: bash -c "java -Xdebug -Xrunjdwp:transport=dt_socket,server=y,suspend=n,address=8765 -Dworking.directory=/root/ -jar james-server.jar"
```
Comment (mbaechler) [inline dockerfiles/run/docker-compose.yml]: maybe we could handle the debug part into the james Dockerfile and enable it only when env DEBUG is set to 1 ?

File: dockerfiles/run/docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+      - "1465:465"
+      - "1587:587"
+      - "1993:993"
+      - "8765:8765"
+
+  elasticsearch:
+    image: elasticsearch:2.2.1
+
+  cassandra:
+    image: cassandra:2.2.3
```
Comment (mbaechler) [inline dockerfiles/run/docker-compose.yml]: I'll add a task about porting to cassandra 3

File: README.txt (hunk 2 above)
Comment (aduprat) [inline README.txt]: OK

File: README.txt (hunk 3 above)
Comment (aduprat) [inline README.txt]: OK

File: README.txt (hunk 4 above)
Comment (aduprat) [inline README.txt]: OK

File: dockerfiles/run/docker-compose.yml (hunk 5 above)
Comment (aduprat) [inline dockerfiles/run/docker-compose.yml]: We don't need that yet.

File: dockerfiles/run/docker-compose.yml (hunk 6 above)
Comment (aduprat) [inline dockerfiles/run/docker-compose.yml]: :+1:

**Pull request merged**